internal/cloud/backend_state.go

```diff
@@ -81,19 +108,59 @@ func (r *remoteClient) Put(state []byte) error {
 
 	// Create the new state.
 	_, err = r.client.StateVersions.Create(ctx, r.workspace.ID, options)
+
 	if err != nil {
-		r.stateUploadErr = true
-		return fmt.Errorf("Error uploading state: %v", err)
+		r.stateUpload.hasErrored = true
+		return fmt.Errorf("error uploading state: %v", err)
 	}
 
 	return nil
 }
 
+func getSchemas(ctxOpts *terraform.ContextOpts, services *disco.Disco, state *states.State) (*terraform.Schemas, error) {
+	var schemas *terraform.Schemas // to get our schemas we need a *terraform.Context, a *configs.Config and *states.State
+
+	if ctxOpts == nil {
+		ctxOpts = new(terraform.ContextOpts)
+	}
+
+	// Get our context
+	tfCtx, ctxDiags := terraform.NewContext(ctxOpts)
+	if ctxDiags.HasErrors() {
+		return schemas, fmt.Errorf("error uploading state to Terraform Cloud: %w", ctxDiags.Err())
```

Comment:
I know `getSchemas` is part of the process of uploading state, but we already mention that in the wrapping exception (Line 85) -- it's probably more helpful to be more specific in each of these error returns... like "error creating terraform context" or  "error loading config" and let the wrapping error mention the state upload.

Reply:
No worries, those repetitive errors (`error uploading state to Terraform Cloud: error uploading state to Terraform Cloud`) were not going to be part of the PR once I open it to review, but I can push that change now.